### ISSUE TITLE
Cannot perform touch events on non-focused monitor

### ISSUE THREAD

**Issue by Thesola10** [paperwm/PaperWM#655]
**Describe the bug**
GNOME is capable of handling multiple monitors, each with their own touch surface. When PaperWM is active, touch screen events are discarded for all apps on displays other than the currently focused one, and I need to move the mouse to the other display to use the touch screen on it.

**To Reproduce**
1. Connect two touch capable displays
2. Open windows on both displays with PaperWM active
3. Move the mouse to one display
4. Try to touch the other display
5. Nothing happens

**Expected behavior**
The other window reacts properly to touch events

**System information:**

```
Distribution: Arch Linux
GNOME Shell 45.0
Display server: Wayland
PaperWM branch/tag: develop
PaperWM commit: c0ce6ab65bbf72bfc59579eb24a119060d68b581
Enabled extensions:
- drive-menu@gnome-shell-extensions.gcampax.github.com
- trayIconsReloaded@selfmade.pl
- arcmenu@arcmenu.com
- nightthemeswitcher@romainvigier.fr
- paperwm@paperwm.github.com
- dash-to-panel@jderose9.github.com
- quick-settings-tweaks@qwreey
- expandable-notifications@kaan.g.inam.org
- quick-settings-avatar@d-go
- caffeine@patapon.info
- reboottouefi@ubaygd.com
- guillotine@fopdoodle.net
- quicksettings-audio-devices-renamer@marcinjahn.com
- gsconnect@andyholmes.github.io
- vertical-workspaces@G-dH.github.com
- mediacontrols@cliffniff.github.com
```

**Additional context**
Native GNOME without PaperWM lets touch events through to unfocused monitors.

**Comment by Thesola10** [comment]
This might also occur with a single touch monitor and a non-touch monitor, if the mouse is in the non-touch monitor.

**Comment by jtaala** [comment]
This is likely because of the StackOverlay actor that gets enabled on monitors where the mouse isn't.

See this thread where I talk a bit about this https://github.com/peterfajdiga/karousel/issues/3

It's basically how we do a "per monitor workspace".

**Comment by jtaala** [comment]
Uncomment these lines to see the overlay (that's most likely blocking your touch events):

https://github.com/paperwm/PaperWM/blob/release/stackoverlay.js#L282-L283

**Comment by Thesola10** [comment]
So basically, hooking `StackOverlay`'s `TOUCH_BEGIN` to "switch monitors as if the mouse entered it" could work?

This would also mean, check that if the mouse starts moving again on a monitor that was defocused by a touch event, the switch triggers again properly

**Comment by jtaala** [comment]
In principle yes - making stackoverlay reactive to touch should cause that monitor space to `select` (make it active).

Also, yes, the mouse movement should trigger the mouse-monitor to become active.

**Comment by Thesola10** [comment]
hm, regression seems to have happened at some point during #751 merge, or I did not test focus behavior properly.

**Comment by Thesola10** [comment]
Closed (again, hopefully for good 😅) by #759
